src/types/datetime.rs

```diff
@@ -160,6 +164,16 @@ pub trait PyTimeAccess {
     fn get_fold(&self) -> bool;
 }
 
+/// Trait for accessing the components of a struct containing a tzinfo.
+pub trait PyTzInfoAccess {
+    /// Returns the tzinfo (which may be None).
+    ///
+    /// Implementations should conform to the upstream documentation:
+    /// <https://docs.python.org/3/c-api/datetime.html#c.PyDateTime_DATE_GET_TZINFO>
+    /// <https://docs.python.org/3/c-api/datetime.html#c.PyDateTime_TIME_GET_TZINFO>
+    fn get_tzinfo(&self) -> Option<&PyTzInfo>;
```

Comment:
Because of the change to the ffi layer I guess this will now always be `Some` unless the Python `None` is handled specially.

Reply:
Thanks, fixed. But I wonder if we really need to do the none check twice, once on underlaying layer and once on rust layer, I think might be better if we can just do the check once.